recipes/folly/all/patches/0007-allow-builtins.patch

```diff
@@ -1,13 +1,128 @@
 diff --git a/folly/portability/Builtins.h b/folly/portability/Builtins.h
-index 971cb88..102fda1 100644
+index 971cb8819..e68de4456 100644
 --- a/folly/portability/Builtins.h
 +++ b/folly/portability/Builtins.h
-@@ -41,7 +41,7 @@ FOLLY_ALWAYS_INLINE void __builtin___clear_cache(char* begin, char* end) {
+@@ -41,7 +41,6 @@ FOLLY_ALWAYS_INLINE void __builtin___clear_cache(char* begin, char* end) {
    }
  }
  
 -#if !defined(_MSC_VER) || (_MSC_VER < 1923)
-+#if !defined(_MSC_VER) || (_MSC_VER < 1930)
  FOLLY_ALWAYS_INLINE int __builtin_clz(unsigned int x) {
    unsigned long index;
    return int(_BitScanReverse(&index, (unsigned long)x) ? 31 - index : 32);
+@@ -93,7 +92,6 @@ FOLLY_ALWAYS_INLINE int __builtin_ctzll(unsigned long long x) {
+   return int(_BitScanForward64(&index, x) ? index : 64);
+ }
+ #endif
+-#endif // !defined(_MSC_VER) || (_MSC_VER < 1923)
+ 
+ FOLLY_ALWAYS_INLINE int __builtin_ffs(int x) {
+   unsigned long index;
+@@ -119,15 +117,12 @@ FOLLY_ALWAYS_INLINE int __builtin_popcount(unsigned int x) {
+   return int(__popcnt(x));
+ }
+ 
+-#if !defined(_MSC_VER) || (_MSC_VER < 1923)
+ FOLLY_ALWAYS_INLINE int __builtin_popcountl(unsigned long x) {
+   static_assert(sizeof(x) == 4, "");
+   return int(__popcnt(x));
+ }
+-#endif // !defined(_MSC_VER) || (_MSC_VER < 1923)
+ #endif
+ 
+-#if !defined(_MSC_VER) || (_MSC_VER < 1923)
+ #if defined(_M_IX86)
+ FOLLY_ALWAYS_INLINE int __builtin_popcountll(unsigned long long x) {
+   return int(__popcnt((unsigned int)(x >> 32))) +
+@@ -138,7 +133,6 @@ FOLLY_ALWAYS_INLINE int __builtin_popcountll(unsigned long long x) {
+   return int(__popcnt64(x));
+ }
+ #endif
+-#endif // !defined(_MSC_VER) || (_MSC_VER < 1923)
+ 
+ FOLLY_ALWAYS_INLINE void* __builtin_return_address(unsigned int frame) {
+   // I really hope frame is zero...
+-- 
+
+diff --git a/folly/portability/Builtins.h b/folly/portability/Builtins.h
+index e68de4456..30caf4003 100644
+--- a/folly/portability/Builtins.h
++++ b/folly/portability/Builtins.h
+@@ -16,7 +16,7 @@
+ 
+ #pragma once
+ 
+-#if defined(_WIN32) && !defined(__clang__)
++#if defined(_WIN32) && !defined(__MINGW32__) && !defined(__clang__)
+ #include <assert.h>
+ #include <folly/Portability.h>
+ #include <intrin.h>
+-- 
+
+diff --git a/folly/portability/Builtins.h b/folly/portability/Builtins.h
+index 30caf4003..e8ef97266 100644
+--- a/folly/portability/Builtins.h
++++ b/folly/portability/Builtins.h
+@@ -22,6 +22,14 @@
+ #include <intrin.h>
+ #include <stdint.h>
+ 
++// MSVC had added support for __builtin_clz etc. in 16.3 (1923) but it will be
```

Comment:
please add a link to the folly issue you mentioned previously!

Reply:
Done